trino/client.py

```diff
@@ -686,7 +686,14 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
         if self.cancelled:
             raise exceptions.TrinoUserError("Query has been cancelled", self.query_id)
 
-        response = self._request.post(self._sql, additional_http_headers)
+        if self._experimental_python_types:
+            http_headers = {constants.HEADER_CLIENT_CAPABILITIES: 'PARAMETRIC_DATETIME'}
```

Comment:
Another naive question - how come this doesn't require any other changes in how the client parses results from the server?

Reply:
What's the current behavior, without setting this header?

Reply:
The server just starts sending back the precision in the type signatures and the values have additional precision to match.

Reply:
The client capabilities doesn't seem to be documented anywhere, but I found this comment in the server: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/server/protocol/QueryResultRows.java#L355
So looks like it was rounded before in the server, and now it'll get truncated in the client, but still with higher precision (6 vs 3). So it's an improvement.

Reply:
All the other Trino clients I've seen (JDBC, ODBC, Go) are all passing `X-Trino-Client-Capabilityes: PARAMETRIC_DATETIME` to Trino when executing a query. The Python library is the only one that doesn't. Unfortunately, Python datetimes only support a 6-digit precision max.

Reply:
@nineinchnick as far as how come no other change is required on how the client parses the results from the server, it's because in my last PR I made sure that the code worked with either type of result.